src/3d/shaders/mesh/mesh.frag

```diff
@@ -185,6 +193,37 @@ vec4 colorRamp()
     return vec4(colorRampResult,1);
 }
 
+float arrows()
+{
+    ivec2 size=textureSize(arrowsGridTexture,0);
+
+    float posX=(fs_in.worldPosition.x-arrowsMinCorner.x)/arrowsSpacing;
+    float posY=(-fs_in.worldPosition.z+arrowsMinCorner.y)/arrowsSpacing;
+    int gridPosX=int(posX+0.5);
+    int gridPosY=int(posY+0.5);
+
+    if (gridPosX<0 || gridPosX>=size.x || gridPosY<0 || gridPosY>=size.y)
+        return 0.0;
+
+    vec2 textureGridPosition=vec2((gridPosX+0.5)/size.x,(gridPosY+0.5)/size.y);
```

Comment:
I am not sure if I understand why we are adding 0.5 here again

Reply:
to get the value on the center of the texel and avoiding interpolation value. But after look at the opengl doc, it seems to exist a function to directly access to the value from texel space coordinates (int, int), instead of normalized texture coordinate . I have to try with.
This is the case too for the color ramp texture.

Reply:
Indeed, textelFetch is more adapted.